.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,49 @@
+name: clang-format (check)
+
+on:
+  pull_request: { branches: ["main"] }
```

Comment:
I didn't run this check on pushes to main: We aren't very strict about this check so I feel like we maybe don't need to track it that carefully.